--- a/Makefile	2022-12-01 12:42:03.053193511 +0100
+++ b/Makefile	2022-12-01 12:45:31.777033969 +0100
@@ -77,7 +77,7 @@
 	$(CXX) $(CXXFLAGS) -std=c++17 -flto -O3 -c src/crypto/*.cpp
 endif
 # Create a static library (try windows, then unix)
-	lib.exe /out:uSockets.a *.o || $(AR) rvs uSockets.a *.o
+	lib.exe /out:uSockets.a *.o || $(AR) rvs libuSockets.a *.o
 
 # BoringSSL needs cmake and golang
 .PHONY: boringssl
